### ISSUE TITLE
Error in Go routine: duplicate metrics collector registration attempted. 502 Gateway error on every request.

### ISSUE THREAD

**Issue by wojciech11** [containous/traefik#1146]
### What version of Traefik are you using (`traefik version`)?
1.2.0 RC1

### What is your environment & configuration (arguments, toml...)?
- I use the kubernetes integration

- args:

          args:
        - --web
        - --kubernetes
        - --configFile=/etc/traefik/traefik.toml
        - --logLevel=INFO
        - --web.metrics.prometheus

### What did you do?

I have deleted a pod (the only pod) for myservice. New one was created.

kubectl delete po myservice

### What did you expect to see?
Traffic goes to myservice instance.

### What did you see instead?
502 Bad Gateway.

```
time="2017-02-09T11:12:33Z" level=warning msg="Endpoints not found for default/myservice, falling back to Service ClusterIP"
time="2017-02-09T11:12:33Z" level=error msg="Error in Go routine: duplicate metrics collector registration attempted"
goroutine 16 [running]:
runtime/debug.Stack(0x4e808f, 0xc420014440, 0x1d9a271)
	/usr/local/go/src/runtime/debug/stack.go:24 +0x79
runtime/debug.PrintStack()
	/usr/local/go/src/runtime/debug/stack.go:16 +0x22
github.com/containous/traefik/safe.defaultRecoverGoroutine(0x1ba12c0, 0xc420a340a0)
	/go/src/github.com/containous/traefik/safe/routine.go:147 +0x7d
github.com/containous/traefik/safe.GoWithRecover.func1.1(0x1e96ec8)
	/go/src/github.com/containous/traefik/safe/routine.go:138 +0x57
panic(0x1ba12c0, 0xc420a340a0)
	/usr/local/go/src/runtime/panic.go:458 +0x243
github.com/containous/traefik/vendor/github.com/prometheus/client_golang/prometheus.(*Registry).MustRegister(0xc42041e940, 0xc420409c10, 0x1, 0x1)
	/go/src/github.com/containous/traefik/vendor/github.com/prometheus/client_golang/prometheus/registry.go:404 +0xa4
github.com/containous/traefik/vendor/github.com/prometheus/client_golang/prometheus.MustRegister(0xc420409c10, 0x1, 0x1)
	/go/src/github.com/containous/traefik/vendor/github.com/prometheus/client_golang/prometheus/registry.go:152 +0x53
github.com/containous/traefik/vendor/github.com/go-kit/kit/metrics/prometheus.NewCounterFrom(0x0, 0x0, 0x0, 0x0, 0x1d99adb, 0x16, 0x1dd9e80, 0x48, 0xc420189530, 0xc4208a9fa0, ...)
	/go/src/github.com/containous/traefik/vendor/github.com/go-kit/kit/metrics/prometheus/prometheus.go:24 +0xe7
github.com/containous/traefik/middlewares.NewPrometheus(0xc420b01740, 0x11, 0xc4202fd460, 0xc4201894d0)
	/go/src/github.com/containous/traefik/middlewares/prometheus.go:42 +0x209
main.(*Server).loadConfig(0xc420058d00, 0xc4201888a0, 0xa, 0x100, 0x0, 0x0, 0x0, 0x0, 0xc42022210b, 0x4, ...)
	/go/src/github.com/containous/traefik/server.go:719 +0x195f
main.(*Server).listenConfigurations(0xc420058d00, 0xc42079cee0)
	/go/src/github.com/containous/traefik/server.go:286 +0x350
main.(*Server).Start.func2(0xc42079cee0)
	/go/src/github.com/containous/traefik/server.go:103 +0x34
github.com/containous/traefik/safe.(*Pool).Go.func1()
	/go/src/github.com/containous/traefik/safe/routine.go:77 +0x3a
github.com/containous/traefik/safe.GoWithRecover.func1(0x1e96ec8, 0xc420260b40)
	/go/src/github.com/containous/traefik/safe/routine.go:141 +0x4d
created by github.com/containous/traefik/safe.GoWithRecover
	/go/src/github.com/containous/traefik/safe/routine.go:142 +0x49
time="2017-02-09T11:12:34Z" level=warning msg="Endpoints not found for default/myservice, falling back to Service ClusterIP"
time="2017-02-09T11:12:34Z" level=warning msg="Endpoints not found for default/myservice, falling back to Service ClusterIP"

```

**Comment by dtomcej** [comment]
What happens when you run `kubectl get endpoints myservice`? Is your new pod listed there?

The kubernetes client is stating that the new pod is not registering as a ready endpoint.

does the same thing happen if you use a simple nginx pod instead of your app pod?

**Comment by wojciech11** [comment]
The service is there. I did not delete the SVC, just the pods that serve it.

**Comment by wojciech11** [comment]
Any recreation of pods on 1.2.0 RC1 causes the prometheus integration issue. The traefik is not able to serve traffic to newly created pods. I have just upgraded my k8s cluster. All the pods, that were created after traefik pods, casues 502 when I try to connect to them.

**Comment by timoreimann** [comment]
Fixed by #1258.
